src/main/java/org/sagebionetworks/web/client/SynapseJavascriptClient.java

```diff
@@ -130,7 +139,7 @@ public void onResponseReceived(Request request,
 					if (statusCode == SC_OK) {
 						try {
 							JSONObjectAdapter jsonObject = jsonObjectAdapter.createNew(response.getText());
-							callback.onSuccess(jsonObject);
+							callback.onSuccess(jsFactory.newInstance(responseType, jsonObject));
```

Comment:
Throw checked exception in newInstance if type not recognized.  Then send error to original callback.